core/sql/sqlcat/TrafDDLdesc.h

```diff
@@ -1105,6 +1129,114 @@ class TrafViewDesc : public TrafDesc {
   char filler[22];
 };
 
+// --------------------------- privilege descriptors ---------------------------
+// The privilege descriptors are organized as follows:
+//   privDesc - a TrafPrivDesc containing the privGrantees
+//   privGrantees - a TrafPrivGranteeDesc descriptor containing:
+//     grantee - int32 value for each user granted a privilege directly or
+//               through a role granted to the user
+//     objectBitmap  - a TrafPrivBitmapDesc containing granted object privs
+//                     summarized across all grantors
+//     columnBitmaps - list of TrafPrivBitmapDesc, one per colummn, containing 
+//                     granted column privs, summarized across all grantors
+//   priv_bits desc - a TrafPrivBitmapDesc descriptor containing:
+//     privBitmap - bitmap containing granted privs such as SELECT
+//     privWGO - bitmap containing WGO for associated grant (privBitmap)
+//     columnOrdinal - column number for bitmap, for objects, column
+//                     number is not relavent so it is set to -1.
+//   column bits - list of TrafPrivBitmapDesc
+class TrafPrivDesc : public TrafDesc {
+public:
+  TrafPrivDesc() : TrafDesc(DESC_PRIV_TYPE)
+  {}
+
+  // ---------------------------------------------------------------------
+  // Redefine virtual functions required for Versioning.
+  //----------------------------------------------------------------------
+  virtual unsigned char getClassVersionID()
+  {
+    return 1;
+  }
+
+  virtual void populateImageVersionIDArray()
+  {
+    setImageVersionID(0,getClassVersionID());
+  }
+
+  virtual short getClassSize()      { return (short)sizeof(TrafPrivDesc); }
+
+  virtual Long pack(void *space);
```

Comment:
I know it has nothing to do with your code, but I wonder why we use "Long" for the return from pack, and "Lng32" for the return from unpack.